src/Entity/EntityId.php

```diff
@@ -17,6 +18,30 @@
 
 	protected $serialization;
 
+	const PATTERN = '/^:?(\w+:)*[^:]+\z/';
+
+	/**
+	 * @param string $serialization
+	 */
+	public function __construct( $serialization ) {
+		$this::assertValidSerialization( $serialization );
+		$this->serialization = self::normalizeIdSerialization( $serialization );
+	}
+
+	private static function assertValidSerialization( $serialization ) {
+		if ( !is_string( $serialization ) ) {
+			throw new InvalidArgumentException( '$serialization must be a string' );
+		}
+
+		if ( $serialization === '' ) {
```

Comment:
I find it best practice to merge the first two trivial assertions into a single "must be a non-empty string".